galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerContainerResourceMonitor.java

```diff
@@ -0,0 +1,163 @@
+
+        for (String engine : dockerEngines.keySet()) {
+            List<String> containers = getOrphanedContainers(engine, dockerEngines.get(engine));
+            logger.info("Engine " + engine + " has " + containers.size() + " orphaned containers found");
+            if (containers.size() > 0) {
+                killContainers(containers, dockerEngines.get(engine));
+            }
+        }
+        logger.info("Docker container resouce check Finished");
+    }
+
+    /**
+     * Stops and removed containers from an engines
+     * @param containers
+     * @param client
+     */
+    private void killContainers(List<String> containers, IHttpClient client) {
+        logger.info("Shutting down oprhaned containers");
+        try {
+            for (String id : containers) {
+                HttpClientResponse<String> resp = client.deleteText("/containers/"+id+"?force=true");
+                if (resp.getStatusCode() != 204) {
+                    logger.error("Something went wrong when removing container: " + resp.getStatusLine());
+                }
+            }
+        } catch (HttpClientException e) {
+            logger.error("Failed to kill containers.", e);
+        }        
+    }
+
+    /**
+     * Looks at all containers on a Engine, locates Galasa specifics and ensures they have a decicated slot
+     * @param engine
+     * @param client
+     * @return
+     */
+    private List<String> getOrphanedContainers(String engine, IHttpClient client) {
+        List<String> orphanedContainers = new ArrayList<>();
+        try {
+            HttpClientResponse<String> resp = client.getText("/containers/json");
+            if (resp.getStatusCode() != 200) {
+                logger.error("Something went wrong when retrieving containers: " + resp.getStatusLine());
+            }
+
+            DockerContainerJSON[] activeContainers = gson.fromJson(resp.getContent(), DockerContainerJSON[].class);
+            for (DockerContainerJSON container : activeContainers) {
+                String runName = parseRunId(container.getNames());
+                // Other non Galasa pod
+                if (runName == null) {
+                    continue;
+                }
+
+                Collection<String> activeRunNames = dss.getPrefix("engine."+engine+".slot").values();
+                if (!activeRunNames.contains(runName)) {
+                    orphanedContainers.add(container.getId());
+                }
+            }
+        } catch (HttpClientException | FrameworkException e) {
+            logger.error("Failed to get containers.", e);
+        }
+        return orphanedContainers;
+    }
+
+    /**
+     * Looks at CPS for the docker Engines available
+     * @return
+     */
+    private Map<String, IHttpClient> getDockerEngines() {
+        HashMap<String, IHttpClient> engines = new HashMap<>();
+        try {
+            String[] enginesTags = cps.getPrefixedProperties("engines").get("engines").split(",");
```

Comment:
not sure why you want to do 2 gets.      get the actual property,  not prefixed ones!